packages/playwright/src/reporters/blob.ts

```diff
@@ -107,17 +107,15 @@ export class BlobReporter extends TeleReporterEmitter {
   }
 
   private async _prepareOutputFile() {
-    let outputFile = reportOutputFileFromEnv();
-    if (!outputFile && this._options.outputFile)
-      outputFile = path.resolve(this._options.configDir, this._options.outputFile);
-    // Explicit `outputFile` overrides `outputDir` and `fileName` options.
-    if (!outputFile) {
-      const reportName = this._options.fileName || process.env[`PLAYWRIGHT_BLOB_OUTPUT_NAME`] || this._defaultReportName(this._config);
-      const outputDir = resolveReporterOutputPath('blob-report', this._options.configDir, this._options.outputDir ?? reportOutputDirFromEnv());
-      if (!process.env.PWTEST_BLOB_DO_NOT_REMOVE)
-        await removeFolders([outputDir]);
-      outputFile = path.resolve(outputDir, reportName);
-    }
+    const { outputFile, outputDir } = resolveOutputFile('BLOB', {
+      ...this._options,
+      default: {
+        fileName: this._defaultReportName(this._config),
+        outputDir: 'blob-report',
+      }
+    })!;
+    if (!process.env.PWTEST_BLOB_DO_NOT_REMOVE)
+      await removeFolders([outputDir!]);
```

Comment:
We should not remove the default `blob-report` dir when `outputFile` option was specified.